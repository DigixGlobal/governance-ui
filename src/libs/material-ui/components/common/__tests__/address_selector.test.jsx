import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AddressSelector from '../address_selector';

const addresses = [{
  name: 'address 1',
  address: 'address test 1',
  keystore: {
    type: { color: 'white' },
  },
}, {
  name: 'address 2',
  address: 'address test 2',
  keystore: {
    type: { color: 'black' },
  },
}, {
  name: 'address 3',
  address: 'address test 4',
  keystore: {
    type: { color: 'orange' },
  },
}];

describe('<AddressSelector />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <AddressSelector
        addresses={addresses}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
