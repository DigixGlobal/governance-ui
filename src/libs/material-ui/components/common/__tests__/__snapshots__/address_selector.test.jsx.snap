// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressSelector /> renders correctly 1`] = `
<DropdownSelector
  items={
    Array [
      Object {
        "address": "address test 1",
        "id": "address test 1",
        "keystore": Object {
          "type": Object {
            "color": "white",
          },
        },
        "name": "address 1",
      },
      Object {
        "address": "address test 2",
        "id": "address test 2",
        "keystore": Object {
          "type": Object {
            "color": "black",
          },
        },
        "name": "address 2",
      },
      Object {
        "address": "address test 4",
        "id": "address test 4",
        "keystore": Object {
          "type": Object {
            "color": "orange",
          },
        },
        "name": "address 3",
      },
    ]
  }
  onChange={null}
  preText={null}
  props={Object {}}
  renderBottom={[Function]}
  renderItem={[Function]}
  renderTrigger={[Function]}
/>
`;
